sentry_streams/sentry_streams/adapters/stream_adapter.py

```diff
@@ -96,6 +98,17 @@ def reduce(
         """
         raise NotImplementedError
 
+    @abstractmethod
+    def router(
+        self,
+        step: Router[RoutingFuncReturnType],
+        stream: Stream,
+    ) -> Stream:
```

Comment:
Is this typehint accurate now? Router can now also return a map of branch names: branches